### PR TITLE
Add resource

### PR DESCRIPTION
**Pull request by TeaBough**
I've created this PR, that serve the same purpose as https://github.com/intljusticemission/react-big-calendar/pull/373. The difference is that the logic of the resource is done directly in the `TimeGrid` instead of creating a `ResourceGrid` which DRY things up and allow to enjoy resource in the week view as well.

### CODE REVIEW

**Comment by jquense** [comment]
🎉 thanks for picking this up!

File: src/TimeGrid.js [hunk 1]
```diff
@@ -174,10 +176,14 @@ export default class TimeGrid extends Component {
 
     let gutterRef = ref => this._gutters[1] = ref && findDOMNode(ref);
 
+    let eventsRendered = resources ?
```
Comment (jquense) [inline src/TimeGrid.js]: `resources && resources.length` ?

File: src/TimeGrid.js [hunk 2]
```diff
@@ -189,22 +195,56 @@ export default class TimeGrid extends Component {
...
+      let daysEvents = events.filter(
+          event => dates.inRange(date,
+              get(event, startAccessor),
+              get(event, endAccessor), 'day')
+      )
+
+      return resources.map((resource, id) => {
+
+        let eventsToDisplay = daysEvents.filter(
+            event => event.resourceId === resource.id
```
Comment (jquense) [inline src/TimeGrid.js]: Like the `events` I don't want to be prescriptive about the shape of the data in `resources` so can we add `accessor` props for these fields like we do with startAccessor, et al?

File: src/TimeGrid.js [hunk 3]
```diff
@@ -189,22 +195,56 @@ export default class TimeGrid extends Component {
...
           />
-
-          {this.renderEvents(range, rangeEvents, this.props.now)}
+          {eventsRendered}
 
         </div>
       </div>
     );
   }
+  renderEventsWithResource(range, events, today, resources) {
```
Comment (jquense) [inline src/TimeGrid.js]: can this method be joined with `renderEvents` I'd be worried the common code will diverge unintentionally

**Comment by XVincentX** [comment]
@TeaBough Is this still happening?

**Comment by TeaBough** [comment]
I couldn't work on it during the french summer. But I think I can work on this this weekend !

**Comment by XVincentX** [comment]
I think this feature would be a significant game changer here 👍

**Comment by TeaBough** [comment]
@XVincentX @jquense I've finally updated this PR

**Comment by jquense** [comment]
Awesome thanks! I'll try and review asap

**Comment by jquense** [comment]
looks great! Can you rebase? The one concern i see here is that with a few resources the rendering hangs for a few seconds as you switch to the tab. I'm not sure what it is that is causing it, it may just be the number of divs being rendered as the grid since I see the same thing with or without any events.

**Comment by TeaBough** [comment]
I've rebased

**Comment by rvignesh89** [comment]
This is great! Exactly what I've been looking for. @jquense any chance that this will be added soon?

**Comment by arthurmmedeiros** [comment]
Hey guys, 
When is this feature going to be added to the master?

**Comment by ksloan** [comment]
Any updates?

**Comment by samlll42-github** [comment]
Looking forward to this as well. Happy to help moving this along if it requires some testing or updates....

**Comment by jquense** [comment]
If anyone wants to help, the last remaining thing is the performance concerns I noted in my comment. If anyone wants to take a crack at optimizing/testing that a bit

**Comment by vongdarakia** [comment]
There's a slight performance issue, but it's a not a big deal for me. It's still way faster than other calendars out there. They often do an http request. I'd really like to see this feature out though. It'd be very helpful! Thanks for all the hard work @jquense and @TeaBough!

File: src/DayColumn.js [hunk 4]
```diff
@@ -64,6 +64,7 @@ class DayColumn extends React.Component {
     dayWrapperComponent: elementType,
     eventComponent: elementType,
     eventWrapperComponent: elementType.isRequired,
+    resource: React.PropTypes.string
```
Comment (frehner) [inline src/DayColumn.js]: Need to remove `React.` here: just `PropTypes.string`

**Comment by frehner** [comment]
Here's a screenshot looking at the Chrome Dev Tool's performance tab when the resource example is created in Storybook. 

screen shot 2017-11-28 at 11 10 50 am

I'm no expert, but it appears that there isn't anything _new_ per se that this PR introduces that reduces performance, it's just the fact that you're now rendering _N_ number of `DayColumn`s (one per resource).  I may be wrong, but if performance is a worry here then we'll either have to figure out a different way to do resources, or go into Day/Time Column components and optimize there.

Thoughts?

**Comment by jquense** [comment]
yeah it may just be the extra columns now. The issue is that it's gonna worse and worse the more resources you have. One solution may be to do something completely different layout wise, like full calendar: https://fullcalendar.io/scheduler/ something a bit more amenable to simpler layout.

another option is to maybe try and virtualize the TimeColumns so it's not rendering extra divs. We can also jsut try and reduce the amount of DOM elements rendered in the first place

**Comment by frehner** [comment]
I'm currently looking at a way of possibly just having one DayColumn component and just using styling to put the events into different columns... does that seem like a reasonable solution?

**Comment by jquense** [comment]
I'm up for whatever works and isnt too hacky :P

**Comment by frehner** [comment]
On second thought, changing the CSS works well when it comes to displaying the events, but would require more work when you consider creating events. Hmm. I'm not really sure what the best way to go about doing this is then, other than what has already been done here.

**Comment by XVincentX** [comment]
Will this find its way in the master somehow? @TeaBough

**Comment by TeaBough** [comment]
@XVincentX I've rebased with master and removed `React.` from `React.PropTypes`. I don't know what else I can do to make this PR ready to merge...

**Comment by jquense** [comment]
if we can make CI green i'm ok merging for now. We really need to address the performance issues though

**Comment by TeaBough** [comment]
@jquense done

**Comment by marwej** [comment]
Any thoughts on when next version (including this) will be released? (Many thanks for a great package!)

**Comment by f2net** [comment]
Could you please add the new "resources" API in the documentation?
I'm looking at the source to understand it, but it would be great to have it documented.

Thank you,
Matteo

**Comment by atul-vzn** [comment]
Does anyone have a working example?

Thank you
Atul

**Comment by justoverclockl** [comment]
> Could you please add the new "resources" API in the documentation? I'm looking at the source to understand it, but it would be great to have it documented.
> 
> Thank you, Matteo

any chance to know if this is implemented in 1.8.1? with a sort of documentation? thx!

**Pull request merged**